analytics: Add summary statistics section to page.

laurynmm opened the pull request:
Adds a summary statistics section to the organization `/stats/` page. and adds the first three statistics using the data already there from the charts. Also, reorders the charts on the page and changes HTML/CSS to use Flexbox.

1. Number of users
2. Total number of messages
3. Number of messages in the last 30 days

Part of #20162. Relevant [CZO conversation](https://chat.zulip.org/#narrow/stream/101-design/topic/summary.20stats.20.2320162/near/1441118).

---

**Notes**:
- The first commit is the reordering of the charts and switching to Flexbox.
- The second commit adds the summary statistics section to the HTML and the active users stat. This value matches the last updated count to the related chart.
- The 30 days statistic matches the **Messages sent over time** chart when set to daily and last 30 days, which means the current day is likely a partial sum since this count is updated every hour.

---

**Screenshots and screen captures:**

<details>
<summary>All charts</summary>

![Screenshot from 2022-09-28 17-31-36](https://user-images.githubusercontent.com/63245456/192824591-02457120-83cb-4085-bcc0-c1c5c43f39aa.png)
</details>

<details>
<summary>Summary section</summary>

![Screenshot from 2022-09-28 17-33-08](https://user-images.githubusercontent.com/63245456/192824201-44bf20f2-44ab-4c43-a556-439532e2d22f.png)
</details>

<details>
<summary>Screencast of resizing the window</summary>

[Screencast from 28-09-22 17:32:10.webm](https://user-images.githubusercontent.com/63245456/192824262-7eca51b4-b6fc-481a-b98c-f074739c0223.webm)
</details>

---

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

timabbott commented on static/js/stats/stats.js:
```diff
@@ -785,6 +799,7 @@ function populate_number_of_users(data) {
     // Initial drawing of plot
     draw_or_update_plot(all_time_trace, true);
     $("#all_time_actives_button").addClass("selected");
+    get_total_active_users(data);
```
Would it be cleaner to pass `data.everyone`? It's not 100% clear to me whether this will get total users (or total active in last N days); might be worth a comment explaining that.

It's also possible that we'll want to display both numbers in the stats area; is that easy if we want to do it?

laurynmm replied:
> Would it be cleaner to pass `data.everyone`?

:+1: 

> It's not 100% clear to me whether this will get total users (or total active in last N days); might be worth a comment explaining that.

Yeah, this statistic is users that haven't been deactivated in the database and who aren't bots. So maybe a better name would be `get_total_not_deactivated_users`? Or `get_total_activated_users`? Or `get_total_users` with a note that we exclude deactivated users and bots?

> It's also possible that we'll want to display both numbers in the stats area; is that easy if we want to do it?

We currently track time series for  "active" users in the last day and 15 days, so those stats for the most recent day would be easy to pull out for the summary section.

timabbott commented:
The implementation looks good; posted a comment on structure. Looking at the screenshots, I'm thinking I might prefer the following formatting:
* 1200 total users
* 123 users active in last 30 days
* 120321 total messages sent
* 12312 messages sent in last 30 days

rather than the `field: value` structure, but I'm not sure. One downside is that'd be worse for i18n. Another option would be to do a simple table, where we could do "all time", "last 30 days" as columns, and "messages sent" and "users" -- though explaining the 'active' aspect of the users bit active definition may get complicated there.

alya commented:
I don't know how to fully test on dev (see below), but the screenshots, etc. in the PR description look great to me!
<img width="605" alt="Screen Shot 2022-09-28 at 10 39 02 AM" src="https://user-images.githubusercontent.com/2090066/192851545-ed330014-fb4f-4921-8c2c-8be4f2ba975a.png">

alya commented:
While we're here, can we do something less silly when the data for the charts is missing? Ideally explicitly displaying text that says "Data not available.", but at the very least not showing broken axis labels?

This can be a separate PR, of course.

<img width="1678" alt="Screen Shot 2022-09-28 at 10 39 39 AM" src="https://user-images.githubusercontent.com/2090066/192851988-b2245a17-3615-4162-89c0-7a71dc87ba52.png">

<img width="1055" alt="Screen Shot 2022-09-28 at 10 41 49 AM" src="https://user-images.githubusercontent.com/2090066/192852032-6f4e9d52-8f3b-436d-a20a-1bb984c4178e.png">

laurynmm commented:
> I'm thinking I might prefer the following formatting:
> 
>     * 1200 total users
>     * 123 users active in last 30 days
>     * 120321 total messages sent
>     * 12312 messages sent in last 30 days

Here's a screenshot with that formatting, though I agree that format due to pluralization would be more difficult:

<details>
<summary>Alternate list format</summary>

![Screenshot from 2022-09-29 16-00-36](https://user-images.githubusercontent.com/63245456/193052381-56be6f96-b927-410c-b4c2-9ee30fc97634.png)
</details>

--- 
> Another option would be to do a simple table, where we could do "all time", "last 30 days" as columns, and "messages sent" and "users" -- though explaining the 'active' aspect of the users bit active definition may get complicated there.

We don't have active users for the last 30 days, but we do for the last 15 days. So we could do a column with "last 15 days" and adjust the messages sent statistic. If we want the users, then we'd have to create another count statistic for that.

Also, would we want to make that chart in HTML or use Plotly.js? Would we want the simple list for now and then have the goal of a chart when/if we switch to another graphing library since Plotly.js isn't something we want to continue to work with in the long run?

laurynmm commented:
**Updates**:
- Renamed the user statistic function (per [this comment](https://github.com/zulip/zulip/pull/23085#discussion_r982658207)) and added the active in the last 15 days user stat.
- I did not change the list format, so it's still `field: value` structure (see [this comment](https://github.com/zulip/zulip/pull/23085#issuecomment-1262331730) with screenshots of potential changes).

timabbott commented:
Merged, thanks @laurynmm! I think this is a nice checkpoint.

I'm not sure which of those alternative presentations might be best, and they're all more work than the current version, so I figure we can discuss that and fix forward, in parallel with running this on chat.zulip.org, where we'll be able to see how real data ends up being presented.

One other thing on my mind in this is the word "Active" -- there's two definitions one might imagine in the Zulip context -- non-deactivated accounts, and active in the sense of having actually used the product in a UI (which is what the 15 day actives number captures; I suspect email-only use doesn't get counted).

alya commented:
Yeah, I'm not sure about the format ideas... I think with the current number of bullets, the format in this PR seems fine, but we might end up wanting to structure it more once more fields get added.
